tests/helpers_tests/test_audience.py

```diff
@@ -171,12 +181,92 @@ def test_is_user_in_experiment__evaluates_audience_conditions_leaf_node(self):
     experiment.audienceConditions = '3468206645'
 
     with mock.patch('optimizely.helpers.condition.CustomAttributeConditionEvaluator') as custom_attr_eval:
-      audience.is_user_in_experiment(project_config, experiment, {})
+      audience.is_user_in_experiment(project_config, experiment, {}, self.mock_client_logger)
 
     audience_3468206645 = project_config.get_audience('3468206645')
 
     custom_attr_eval.assert_has_calls([
-        mock.call(audience_3468206645.conditionList, {}),
+        mock.call(audience_3468206645.conditionList, {}, self.mock_client_logger),
         mock.call().evaluate(0),
         mock.call().evaluate(1),
     ], any_order=True)
+
+
+class AudienceLoggingTest(base.BaseTest):
+
+  def setUp(self):
+    base.BaseTest.setUp(self)
+    self.mock_client_logger = mock.MagicMock()
+
+  def test_is_user_in_experiment__with_no_audience(self):
+    experiment = self.project_config.get_experiment_from_key('test_experiment')
+    experiment.audienceIds = []
+    experiment.audienceConditions = []
+
+    with mock.patch('optimizely.logger.reset_logger', return_value=self.mock_client_logger):
+      audience.is_user_in_experiment(self.project_config, experiment, {}, self.mock_client_logger)
+
+    self.mock_client_logger.assert_has_calls([
+      mock.call.debug('Evaluating audiences for experiment "test_experiment": [].'),
+      mock.call.info('Audiences for experiment "test_experiment" collectively evaluated to TRUE.')
+    ])
+
+  def test_is_user_in_experiment__evaluates_audienceIds(self):
+    user_attributes = {'test_attribute': 'test_value_1'}
+    experiment = self.project_config.get_experiment_from_key('test_experiment')
+    experiment.audienceIds = ['11154', '11159']
+    experiment.audienceConditions = None
+    audience_11154 = self.project_config.get_audience('11154')
+    audience_11159 = self.project_config.get_audience('11159')
+
+    with mock.patch('optimizely.helpers.condition.CustomAttributeConditionEvaluator.evaluate',
+                    side_effect=[None, None]),\
+      mock.patch('optimizely.logger.reset_logger', return_value=self.mock_client_logger):
+      audience.is_user_in_experiment(self.project_config, experiment, user_attributes, self.mock_client_logger)
+
+    self.assertEqual(3, self.mock_client_logger.debug.call_count)
+    self.assertEqual(3, self.mock_client_logger.info.call_count)
+
+    self.mock_client_logger.assert_has_calls([
+      mock.call.debug('Evaluating audiences for experiment "test_experiment": ["11154", "11159"].'),
+      mock.call.debug('Starting to evaluate audience "11154" with conditions: "' + audience_11154.conditions + '".'),
```

Comment:
We'll need to change this to
```suggestion
      mock.call.debug('Starting to evaluate audience "11154" with conditions: ' + audience_11154.conditions + '.'),
```
per the suggestion that I've made in enums.py. Or even better, IMO:
```suggestion
      mock.call.debug('Starting to evaluate audience "11154" with conditions: ["and", ["or", ["or", '
                      '{"name": "test_attribute", "type": "custom_attribute", "value": "test_value"}]]].'),
```

(Same thing below)